### ISSUE TITLE
Sparse 2D Slices for Preprocessed Input Images

### ISSUE THREAD

**Issue by uzaymacar** [ivadomed/ms-challenge-2021#27]
Trained a 2D multi-channel U-net model with (i) raw input images (located in `~/duke/projects/ivadomed/tmp_ms_challenge_2021`) and (ii) preprocessed input images (according to the latest preprocessing pipeline, located in `~/duke/projects/ivadomed/tmp_ms_challenge_2021_preprocessed`. The training was done with `ivadomed` and (i) had a best-validation Dice score of `0.35` whereas (ii) had a best-validation Dice score of `0.32`. The config file is pretty similar to `config/baseline.json`, with only modifications done on data and output paths, and contrasts. 

*NOTE*: I realize I'm skipping details regarding the model and its training. Goal is to have a more detailed report dedicated to trained models and performance achieved ahead of Wednesday's meeting; to-be-communicated with a new PR and through Slack!

I am suspicious / surprised that the preprocessed data didn't have a higher performance from the get-go 🤔. One potential issue I have identified is the higher ratio of sparse 2D slices as a result of the preprocessing step. These are expected as we are applying brain + SC masks and effectively maximizing the num. of black pixels. Visualized (via `tensorboard`) below are input slices first from (i) and then from (ii).

**RAW**:
![input_images_raw](https://user-images.githubusercontent.com/33473983/121109804-e9a8ff80-c814-11eb-9952-05d22f804f42.gif)

**PREPROCESSED**:
![input_images_preprocessed](https://user-images.githubusercontent.com/33473983/121109811-eca3f000-c814-11eb-8ed5-ce3223dbfe52.gif)

My concern is that the sparsity might be affecting the performance. The *weak hypothesis* is: noisy slices (e.g. whether they have the skull or just random noise) might be enhancing the performance due to expanding the distribution of the training set.

What do you think?

**Next Steps**:
* Maybe 3D training will help with this case -> train 3D multichannel U-Net on (ii) with `ivadomed`. ✅
* Play around with hyperparameters; maybe it's just a bad training session? (not likely IMO.)
* Play around with the data augmentation strategy; we might need to diversify the training set.
